juju-tutorials/how-to-fixing-attach-volume-to-a-k8s-instance.md

```diff
@@ -0,0 +1,336 @@
+   ```bash
+   unset BRIDGE_INTERFACE_MAPPING  # clear value
+
+   for i in 0 1 2
+   do
+       # create new port
+       export NEW_PORT_$i_ID=$(openstack port create --network $ADMIN_NET data-port | grep ' id ' | cut -d"|" -f3 | tr -d " ")
+       echo "  create new port with id $NEW_PORT_$i_ID"
+    
+       # get ovn chassis ip
+       export OVN_CHASSIS_$i_IP=$(juju status | grep ovn-chassis/$i | tr -s " " | cut -d" " -f5)
+       echo "  found ovn-chassis/$i ip address $OVN_CHASSIS_$i_IP"
+    
+       # get ovn chassis id
+       export OVN_CHASSIS_$i_ID=$(openstack server list | grep -E $OVN_CHASSIS_$i_IP | cut -d"|" -f3 | tr -d " ")
+       echo "  found ovn-chassis/$i id $OVN_CHASSIS_$i_ID"
+    
+       # add port to ovn chassis
+       openstack server add port $OVN_CHASSIS_$i_ID $NEW_PORT_$i_ID
+       echo "  add port $NEW_PORT_$i_ID to server $OVN_CHASSIS_$i_ID"
+    
+       # get port mac address
+       export NEW_PORT_$i_MAC_ADDRESS=$(openstack port show $NEW_PORT_$i_ID | grep mac_address | cut -d"|" -f3 | tr -d " ")
+       echo "  port $NEW_PORT_$i_ID mac address $NEW_PORT_$i_MAC_ADDRESS"
+    
+       # set bridge interface mapping variable
+       export BRIDGE_INTERFACE_MAPPING="$BRIDGE_INTERFACE_MAPPING br-ex:$NEW_PORT_$i_MAC_ADDRESS"
+   done
+   ```
+3. Configure the bridge interface mapping in the `ovn-chassis` application.
+   ```bash
+   juju config ovn-chassis bridge-interface-mappings="$BRIDGE_INTERFACE_MAPPING"
+   ```
+   
+##### final configuration
+
+The final step is to configure juju to use deployed OpenStack as cloud and configure network on OpenStack.
+
+[note status="OpenStack credentials"]
+Don't forget to get credentials for the new OpenStack. (e.g. `source openrcv3_project`)
+[/note]
+
+1. To access the OpenStack cloud is necessary to set up flavors, images, networks and security. Detailed instructions can be found on [openstack-base][OpenStackBase] in the "Accessing the cloud" section.
+    [note status="OpenStack network"]
+    Above Openstack, use these network settings to configure the subnet.
+    ```yaml
+    gateway: 10.5.0.1
+    external CIDR: 10.5.0.0/16
+    floating IP range: 10.5.150.0:10.5.200.254
+    internal CIDR: 192.168.21.0/24
+    nameserver: 10.245.168.6
+    ```
+    [/note]
+
+2. The last thing is to add a new OpenStack cloud to Juju, create new credentials and create a controller. See the [OpenStack-cloud][OpenStackCloud] page for more information.
+    [note status="OpenStack metadata"]
+    It's necessary to create metadata for initializing a Juju cloud environment (creating the controller).
+    ```bash
+    juju metadata generate-image -d ~/simplestreams -i <image_id_or_name> -s <series> -r $OS_REGION_NAME -u $OS_AUTH_URL
+    ```
+    [/note]
+
+### Kubernetes
+
+Deployed charmed kubernetes on top of OpenStack with `charm-openstack-integrator` act as proxy for OpenStack and provide a control interface.
```

Comment:
"Deploy CDK (link to CDK) on top of OpenStack using charm-openstack-integrator. The integrator will act as a proxy between Kubernetes and OpenStack, and it will provide a control interface (eg. creating persistent volumes)."